types.go

```diff
@@ -328,7 +328,7 @@ func (trace *CallTrace) addExp(desc string) {
 	trace.ExpDesc[desc] = false
 }
 
-func (trace *CallTrace) addError(err error) {
+func (trace *CallTrace) addFail(err error) {
```

Comment:
A little inconsistent that addFail method accepts error type. addError was more consistent IMO.

Reply:
the issue with addError is that method has extra logic related to expectation failure while error is secondary